session.rst

```diff
@@ -130,105 +391,303 @@ session metadata files:
-            $filters = $session->get('filters', []);
+This metadata can be used to explicitly expire a session on access::
 
-            // ...
-        }
+    $session->start();
+    if (time() - $session->getMetadataBag()->getLastUsed() > $maxIdleTime) {
+        $session->invalidate();
+        throw new SessionExpired(); // redirect to expired session page
     }
 
-.. deprecated:: 5.3
+It is also possible to tell what the ``cookie_lifetime`` was set to for a
+particular cookie by reading the ``getLifetime()`` method::
 
-    The ``SessionInterface`` and ``session`` service were deprecated in
-    Symfony 5.3. Instead, inject the ``RequestStack`` service to get the session
-    object of the current request.
+    $session->getMetadataBag()->getLifetime();
 
-Stored attributes remain in the session for the remainder of that user's session.
-By default, session attributes are key-value pairs managed with the
-:class:`Symfony\\Component\\HttpFoundation\\Session\\Attribute\\AttributeBag`
-class.
+The expiry time of the cookie can be determined by adding the created
+timestamp and the lifetime.
+
+.. index::
+    single: Session; Database Storage
+
+.. _session-database:
+
+Store Sessions in a Database
+----------------------------
+
+Symfony stores sessions in files by default. If your application is served by
+multiple servers, you'll need to use a database instead to make sessions work
+across different servers.
+
+Symfony can store sessions in all kinds of databases (relational, NoSQL and
+key-value) but recommends key-value databases like Redis to get best
+performance.
+
+Store Sessions in a key-value Database (Redis)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This section assumes that you have a fully-working Redis server and have also
+installed and configured the `phpredis extension`_.
+
+You have two different options to use Redis to store sessions:
 
-.. deprecated:: 5.3
+(1) The first PHP-based option is to configure Redis session handler directly
+    in the server ``php.ini`` file:
 
-    The ``NamespacedAttributeBag`` class is deprecated since Symfony 5.3.
-    If you need this feature, you will have to implement the class yourself.
+.. code-block:: ini
 
-If your application needs are complex, you may prefer to use
-:ref:`namespaced session attributes <namespaced-attributes>` which are managed with the
-:class:`Symfony\\Component\\HttpFoundation\\Session\\Attribute\\NamespacedAttributeBag`
-class. Before using them, override the ``session_listener`` service definition to build
-your ``Session`` object with the default ``AttributeBag`` by the ``NamespacedAttributeBag``:
+    ; php.ini
+    session.save_handler = redis
+    session.save_path = "tcp://192.168.0.178:6379?auth=REDIS_PASSWORD"
+
+(2) The second Symfony-based option is to configure Redis sessions as follows.
+
+First, define a Symfony service for the connection to the Redis server:
```

Comment:
```suggestion
The second option is to configure Redis sessions in Symfony. First, define
a Symfony service for the connection to the Redis server:
```

Reply:
slightly related: it looks like we're missing a high-level API for this in the framework.

Is this intended (e.g. do we actually not recommend storing sessions in a db at all) or is this a missing DX feature?